tests/test_config.py

```diff
@@ -753,3 +753,42 @@ def test_specifiers(tmp_path: Path) -> None:
 
 def test_deterministic() -> None:
     assert MkosiConfig.default() == MkosiConfig.default()
+
+
+def test_environment(tmp_path: Path) -> None:
+    d = tmp_path
+
+    (d / "mkosi.conf").write_text(
+        """\
+        [Content]
+        Environment=TestValue2=300
+                    TestValue3=400
+        EnvironmentFiles=other.env
+        """
+    )
+
+    (d / "mkosi.env").write_text(
+        """\
+        TestValue1=90
+        """
+    )
+
+    (d / "other.env").write_text(
+        """\
+        TestValue1=100
+        TestValue2=200
+        """
+    )
+
+    with chdir(d):
+        _, [config] = parse_config()
+
+        expected = {
+            "TestValue1": "100",
+            "TestValue2": "300",
+            "TestValue3": "400"
```

Comment:
To make this a bit clearer
```suggestion
            "TestValue1": "100", # from other.env
            "TestValue2": "300", # from mkosi.conf
            "TestValue3": "400", # from mkosi.conf
```

This also shows that this doesn't test a setting from `mkosi.env` surviving, which would be nice.